### PR TITLE
[Y2K-353] Refactor contact banner...

### PR DESCRIPTION
**Pull request by jzila**
... to support reusing state and dispatch properties in multiple components

cc @keybase/y2ksquad

### CODE REVIEW

**Comment by jzila** [comment]
This is ready for review. I'll make another PR to actually add the row.

File: shared/team-building/container.tsx [hunk 1]
```diff
@@ -172,12 +181,18 @@ const debouncedSearch = debounce(
 const mapDispatchToProps = (dispatch: Container.TypedDispatch, {namespace, teamname}: OwnProps) => ({
   _onAdd: (user: User) => dispatch(TeamBuildingGen.createAddUsersToTeamSoFar({namespace, users: [user]})),
   _onCancelTeamBuilding: () => dispatch(TeamBuildingGen.createCancelTeamBuilding({namespace})),
+  _onImportContactsPermissionsGranted: () =>
+    dispatch(SettingsGen.createRequestContactPermissions({thenToggleImportOn: true})),
+  _onImportContactsPermissionsNotGranted: () =>
+    dispatch(SettingsGen.createEditContactImportEnabled({enable: true})),
```
Comment (buoyad) [inline shared/team-building/container.tsx]: Should this and `...PermissionsGranted` above be swapped? AIUI if permissions are `granted` currently => `requestContactPermissions`, we should do that if not granted.

File: shared/team-building/index.tsx [hunk 2]
```diff
@@ -94,13 +97,21 @@ const ContactsBanner = ({onRedoSearch, onRedoRecs}: {onRedoSearch: () => void; o
 
   // Ensure that we know whether contacts are loaded, and if not, that we load
   // the current config setting.
-  if (contactsImported === null) {
-    dispatch(SettingsGen.createLoadContactImportEnabled())
-    return null
-  }
+  React.useEffect(() => {
+    if (contactsImported === null) {
+      onContactsNotYetImported()
```
Comment (buoyad) [inline shared/team-building/index.tsx]: Rename to `onLoad...`? it sounds like something different currently.

File: shared/team-building/container.tsx (hunk 1 above)
Comment (jzila) [reply shared/team-building/container.tsx]: 🤦‍♀️

**Pull request merged**